test/e2e/app-dir/app/index.test.ts

```diff
@@ -1788,4 +1791,33 @@ describe('app dir - basic', () => {
       })
     }
   })
+
+  // this one comes at the end to not change behavior from above
+  // assertions with compile mode specifically
+  if (process.env.NEXT_EXPERIMENTAL_COMPILE) {
```

Comment:
can we have a new test suite to do this? the `app-dir/app` test suite used to contain some basic tests but evolve too much by testing a lot of new feature. would be easier to locate the env replacement test if it has a standalone fixture

Reply:
added in this suite since it already runs `next build --experimental-build-mode=compile` for the `experimental-compile.test.ts` suite and we need to run `generate` after that if we expand these more we prob want to break out to it's own fixture though so left a note